### PR TITLE
*: update Prometheus / promql-engine dependency with fix + add test case for pushed down subquery

### PR DESCRIPTION
**Pull request by GiedriusS**
Repro for https://github.com/thanos-io/thanos/issues/5600.

Signed-off-by: Giedrius Statkevičius <giedrius.statkevicius@vinted.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -246,6 +246,8 @@ replace (
 	// Required by Cortex https://github.com/cortexproject/cortex/pull/3051.
 	github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-20180902122335-24332e2d58ab
 
+	github.com/prometheus/prometheus => github.com/vinted/prometheus v1.8.2-0.20220817132046-4af7e615cfb4
```
Comment (GiedriusS) [inline go.mod:249]: Temporary fork until https://github.com/prometheus/prometheus/pull/11163 is figured out.

**Review by fpetkovski — DISMISSED** [review]
Thanks, lgtm. 

We can also wait a few days see if your upstream patch gets merged in Prometheus.

**Comment by fpetkovski** [comment]
Hm, is the test failure relevant?

**Comment by GiedriusS** [comment]
> Hm, is the test failure relevant?

Yeah, something went wrong. Looking into it

**Comment by yeya24** [comment]
This contains the new OOO samples feature in TSDB?

**Comment by douglascamata** [comment]
FYI Prometheus has tagged `v0.29.0` just few mins ago (I was about to open a PR).

**Comment by GiedriusS** [comment]
Feel free to update it in a separate PR as I'll be on vacation for a bit

**Comment by matej-g** [comment]
Looks like this is the culprit, because of change here https://github.com/jesusvazquez/prometheus/blob/c785aa0b69c2aab6966209f91d5294fb993e979e/tsdb/wal/wal.go#L282 coming from https://github.com/prometheus/prometheus/pull/11075:
```
panic: duplicate metrics collector registration attempted

goroutine 4571 [running]:
github.com/prometheus/client_golang/prometheus.(*wrappingRegisterer).MustRegister(0xc0000acc60, {0xc000305ea0?, 0x7, 0x0?})
	/home/circleci/go/pkg/mod/github.com/prometheus/client_golang@v1.13.0/prometheus/wrap.go:106 +0x151
github.com/prometheus/prometheus/tsdb/wal.newWALMetrics({0x145ef58, 0xc0000acc60})
	/home/circleci/go/pkg/mod/github.com/prometheus/prometheus@v0.38.1-0.20221003141934-f7a7b18cdcca/tsdb/wal/wal.go:238 +0x64e
github.com/prometheus/prometheus/tsdb/wal.NewSize({0x1457080, 0xc000120500}, {0x145fb28?, 0xc002d22660}, {0xc000626800, 0x3b}, 0x8000000, 0x0)
	/home/circleci/go/pkg/mod/github.com/prometheus/prometheus@v0.38.1-0.20221003141934-f7a7b18cdcca/tsdb/wal/wal.go:282 +0x2e5
github.com/prometheus/prometheus/tsdb.open({0xc000626680, 0x37}, {0x1457080, 0xc000120500}, {0x145fb28?, 0xc002d22660}, 0xc000700480, {0xc000048e60, 0xa, 0xa}, ...)
	/home/circleci/go/pkg/mod/github.com/prometheus/prometheus@v0.38.1-0.20221003141934-f7a7b18cdcca/tsdb/db.go:750 +0x94e
github.com/prometheus/prometheus/tsdb.Open({0xc000626680, 0x37}, {0x1457080, 0xc000120500}, {0x145fb28, 0xc002d22660}, 0x28?, 0x43d147?)
	/home/circleci/go/pkg/mod/github.com/prometheus/prometheus@v0.38.1-0.20221003141934-f7a7b18cdcca/tsdb/db.go:613 +0x9b
github.com/thanos-io/thanos/pkg/receive.(*MultiTSDB).startTSDB(0xc00252c1e0, {0x1457080, 0xc000120500}, {0x113bf28, 0x3}, 0x0?)
	/home/circleci/project/pkg/receive/multitsdb.go:448 +0x42e
github.com/thanos-io/thanos/pkg/receive.(*MultiTSDB).getOrLoadTenant.func1()
	/home/circleci/project/pkg/receive/multitsdb.go:510 +0x4d
created by github.com/thanos-io/thanos/pkg/receive.(*MultiTSDB).getOrLoadTenant
	/home/circleci/project/pkg/receive/multitsdb.go:509 +0x32b
```

**Pull request closed without merging**